cmd/controller/apps_reconciler.go

```diff
@@ -35,5 +39,31 @@ func (r *AppsReconciler) Reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcile.Result{}, err
 	}
 
-	return r.appFactory.NewCRDApp(existingApp, log).Reconcile()
+	force := false
+	crdApp := r.appFactory.NewCRDApp(existingApp, log)
+	r.updateAppRefs(crdApp.GetSecretRefs(), "secret", existingApp)
+	r.updateAppRefs(crdApp.GetConfigMapRefs(), "configmap", existingApp)
+	if r.appUpdateStatus.IsUpdateNeeded(existingApp.Name, existingApp.Namespace) {
+		r.appUpdateStatus.MarkUpdated(existingApp.Name, existingApp.Namespace)
+		force = true
+	}
+
+	return crdApp.Reconcile(force)
+}
+
+func (r *AppsReconciler) updateAppRefs(refNames map[string]struct{}, kind string, app *v1alpha1.App) {
+	// If App is being deleted, remove App from appRefTracker.
+	if app.DeletionTimestamp != nil {
+		for refName := range refNames {
+			r.appRefTracker.RemoveAppFromRefMap(kind, refName, app.Namespace, app.Name)
```

Comment:
we should probably turn this into removing app from all refs. search could be limited by namespace in theory.

Reply:
I changed the func to remove an App from all ref keys by passing in ref names.